src/main/java/frc/robot/Robot.java

```diff
@@ -150,5 +172,10 @@ public void simulationInit() {}
 
   /** This function is called periodically whilst in simulation. */
   @Override
-  public void simulationPeriodic() {}
+  public void simulationPeriodic() {
+    if (ModeConstants.currentMode == frc.robot.constants.ModeConstants.Mode.SIM
+        && ModeConstants.currentMode == frc.robot.constants.ModeConstants.Mode.MAPLESIM) {
```

Comment:
This needs to be `||` rather than `&&`

this could be one cause of issues because this piece of code never runs as it is